cmd/evm/staterunner.go

```diff
@@ -107,19 +107,18 @@ func runStateTest(fname string, cfg vm.Config, jsonOut, dump bool) error {
 			// Run the test and aggregate the result
 			result := &StatetestResult{Name: key, Fork: st.Fork, Pass: true}
 			test.Run(st, cfg, false, rawdb.HashScheme, func(err error, snaps *snapshot.Tree, state *state.StateDB) {
+				root := state.IntermediateRoot(false)
```

Comment:
Previously (before all this) we checked if state != nil before calculating the intermediate root on it. Is this still needed?

Reply:
Well, I'm running this code on a fuzzer, and it does not crash. So in my limited practice-experience, it seems to not be needed. I don't know if the theoretical answer is different

Reply:
You are right -- it can be nil in cases where certain errors have occurred, e..g `UnsupportedForkError`. Then we don't have any state.  I will fix

Reply:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x118 pc=0xc00258]

goroutine 1 [running]:
github.com/ethereum/go-ethereum/core/state.(*StateDB).Finalise(0x0, 0x0)
	github.com/ethereum/go-ethereum/core/state/statedb.go:899 +0x38
github.com/ethereum/go-ethereum/core/state.(*StateDB).IntermediateRoot(0x0, 0xa0?)
	github.com/ethereum/go-ethereum/core/state/statedb.go:951 +0x56
main.runStateTest.func1({0x12f6480, 0xc0001f95c0}, 0x12f8320?, 0xf437a0?)
	github.com/ethereum/go-ethereum/cmd/evm/staterunner.go:110 +0x77
github.com/ethereum/go-ethereum/tests.(*StateTest).Run.func1()
	github.com/ethereum/go-ethereum/tests/state_test_util.go:197 +0x3d
github.com/ethereum/go-ethereum/tests.(*StateTest).Run(0xc00059f660, {{0xc0004e8010?, 0xc00059f478?}, 0x4205a8?}, {{0x1302a88, 0xc000012d50}, 0x0, 0x0, {0x0, 0x0, ...}}, ...)
	github.com/ethereum/go-ethereum/tests/state_test_util.go:205 +0x410
main.runStateTest({0x7ffff4d6e24b?, 0x4?}, {{0x1302a88, 0xc000012d50}, 0x0, 0x0, {0x0, 0x0, 0x0}}, 0x1, ...)
	github.com/ethereum/go-ethereum/cmd/evm/staterunner.go:109 +0x52a
main.stateTestCmd(0xc000270e40)
	github.com/ethereum/go-ethereum/cmd/evm/staterunner.go:77 +0x534
github.com/ethereum/go-ethereum/internal/flags.MigrateGlobalFlags.func2.1(0xc00059fb88?)
	github.com/ethereum/go-ethereum/internal/flags/helpers.go:91 +0x36
github.com/urfave/cli/v2.(*Command).Run(0x1915de0, 0xc000270e40, {0xc0001f8cc0, 0x2, 0x2})
	github.com/urfave/cli/v2@v2.24.1/command.go:271 +0x9eb
github.com/urfave/cli/v2.(*Command).Run(0xc00026e140, 0xc000270100, {0xc0000360a0, 0x5, 0x5})
	github.com/urfave/cli/v2@v2.24.1/command.go:264 +0xc4d
github.com/urfave/cli/v2.(*App).RunContext(0xc0004e6000, {0x12fdfb0?, 0xc00003e100}, {0xc0000360a0, 0x5, 0x5})
	github.com/urfave/cli/v2@v2.24.1/app.go:333 +0x665
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.24.1/app.go:310
main.main()
	github.com/ethereum/go-ethereum/cmd/evm/main.go:227 +0x47

```

Reply:
Panic'd if you specify a non-supported fork (like Istanbul_2)